coordinator/points_writer.go

```diff
@@ -113,12 +113,20 @@ func NewShardMapping(n int) *ShardMapping {
 }
 
 // MapPoint adds the point to the ShardMapping, associated with the given shardInfo.
-func (s *ShardMapping) MapPoint(shardInfo *meta.ShardInfo, p models.Point) {
+func (s *ShardMapping) MapPoint(rp *meta.RetentionPolicyInfo, shardInfo *meta.ShardInfo, p models.Point) (dropped bool) {
+	if (rp != nil) &&
+		(((rp.FutureWriteLimit > 0) && p.Time().After(time.Now().Add(rp.FutureWriteLimit))) ||
+			((rp.PastWriteLimit > 0) && p.Time().Before(time.Now().Add(-rp.PastWriteLimit)))) {
```

Comment:
I'm wondering if we shouldn't move this check into `PointsWriter.MapShards`. Having the check in here could allow us to create shards outside of the write window before we drop the points here. Moving the check into `PointsWriter.MapShards` would prevent the creation of shards outside the write windows.

Reply:
Good idea.